holmes/plugins/toolsets/opensearch/opensearch_logs.py

```diff
@@ -0,0 +1,185 @@
+        except Exception as e:
+            logging.warn("Failed to process opensearch logs fields", exc_info=True)
+            return f"Unexpected error: {str(e)}"
+
+    def get_parameterized_one_liner(self, params) -> str:
+        return "list log documents fields"
+
+
+class LogsSearchQuery(BaseOpenSearchLogsTool):
+    def __init__(self, toolset: "OpenSearchLogsToolset"):
+        super().__init__(
+            name="logs_in_range_search",
+            description="Get logs in a specified time range for an opensearch query",
+            parameters={
+                "query": ToolParameter(
+                    description="An OpenSearch search query. It should be a stringified json, matching opensearch search syntax. "
+                    "The query must contain a 'range' on the @timestamp field. From a given timestamp, until a given timestamp",
+                    type="string",
+                ),
+            },
+            toolset=toolset,
+        )
+        self._cache = None
+
+    def _invoke(self, params: Any) -> str:
+        err_msg = ""
+        try:
+            body = json.loads(params.get("query"))
+            full_query = body
+            full_query["size"] = int(
+                os.environ.get("OPENSEARCH_LOGS_SEARCH_SIZE", "5000")
+            )
+            logging.debug(f"opensearch logs search query: {full_query}")
+            headers = {"Content-Type": "application/json"}
+            headers.update(add_auth_header(self.toolset.config.opensearch_auth_header))
+
+            logs_response = requests.get(
+                url=self.toolset.config.opensearch_logs_url,
+                timeout=180,
+                verify=True,
+                data=json.dumps(full_query),
+                headers=headers,
+            )
+            if logs_response.status_code > 300:
+                err_msg = logs_response.text
+
+            logs_response.raise_for_status()
+            return json.dumps(logs_response.json())
+        except requests.Timeout:
+            logging.warn("Timeout while fetching opensearch logs search", exc_info=True)
+            return f"Request timed out while fetching opensearch logs search {err_msg}"
+        except RequestException as e:
+            logging.warn("Failed to fetch opensearch logs search", exc_info=True)
+            return f"Network error while opensearch logs search {err_msg} {str(e)}"
+        except Exception as e:
+            logging.warn("Failed to process opensearch logs search", exc_info=True)
+            return f"Unexpected error {err_msg}: {str(e)}"
+
+    def get_parameterized_one_liner(self, params) -> str:
+        return f'search logs: query="{params.get("query")}"'
+
+
+class OpenSearchLogsToolset(Toolset):
+    def __init__(self):
+        super().__init__(
+            name="opensearch/logs",
+            description="OpenSearch integration to fetch logs",
+            docs_url="",
+            icon_url="",
```

Comment:
Should `docs_url` and `icon_url` be added now or later? `docs_url` might look like `https://docs.robusta.dev/master/configuration/holmesgpt/toolsets/opensearch_logs.html`

Reply:
fixed